src/Database/Models.php

```diff
@@ -204,14 +206,19 @@ public static function isOwnedBy(Model $authority, Model $model)
      * @param  string|\Closure  $attribute
      * @param  \Illuminate\Database\Eloquent\Model  $authority
      * @param  \Illuminate\Database\Eloquent\Model  $model
+     * @param  bool  $isDefaultAttribute
      * @return bool
      */
-    protected static function isOwnedVia($attribute, Model $authority, Model $model)
+    protected static function isOwnedVia($attribute, Model $authority, Model $model, $isDefaultAttribute = false)
     {
         if ($attribute instanceof Closure) {
             return $attribute($model, $authority);
         }
 
+        if ($isDefaultAttribute && !array_key_exists($attribute, $model->getAttributes())) {
+            return false;
+        }
```

Comment:
Why do we need to add `$isDefaultAttribute` to this check, and pass it all the way down? It makes the whole thing messy. Couldn't we just always do this `array_key_exists` check?

Reply:
Maybe I'm being dense, but you could just create the mapping for ownedVia on those models where user_id is missing no?